### PR TITLE
Retry API requests once if the server returns a 5xx response

### PR DESCRIPTION
**Pull request by mmermerkaya**

### CODE REVIEW

File: packages/installations/src/api/common.ts [hunk 1]
```diff
@@ -75,10 +75,30 @@ export function getHeadersWithAuth(
...
+ * Calls the passed in fetch wrapper and returns the response.
+ * If the returned response has a status of 5xx, re-runs the function once and
+ * returns the response.
+ */
+export async function retryIfServerError(
+  fn: () => Promise<Response>
+): Promise<Response> {
+  const result = await fn();
+
+  if (String(result.status)[0] === '5') {
```
Comment (dwoffinden) [inline packages/installations/src/api/common.ts]: That's pretty gross. Why not just check >=500&&<600?

File: packages/installations/src/api/create-installation.test.ts [hunk 2]
```diff
@@ -104,24 +105,41 @@ describe('createInstallation', () => {
...
       );
-    });
 
-    it('throws a FirebaseError with the error information from the server', async () => {
       await expect(
         createInstallation(appConfig, inProgressInstallationEntry)
       ).to.be.rejectedWith(FirebaseError);
     });
+
+    it('retries once if the server returns a 5xx error', async () => {
```
Comment (dwoffinden) [inline packages/installations/src/api/create-installation.test.ts]: as well as a test that it does retry once, we want tests that it doesn't retry infinitely?

File: packages/installations/src/api/common.ts (hunk 1 above)
Comment (mmermerkaya) [reply packages/installations/src/api/common.ts]: Ehh, it's fine either way, but sure.

File: packages/installations/src/api/create-installation.test.ts (hunk 2 above)
Comment (mmermerkaya) [reply packages/installations/src/api/create-installation.test.ts]: Added tests for `retryIfServerError`. No need to duplicate them in all API functions.

**Pull request merged**